coil-core/src/androidInstrumentedTest/kotlin/coil3/decode/AndroidDecoderTest.kt

```diff
@@ -22,23 +23,55 @@ import coil3.test.utils.decodeBitmapAsset
 import coil3.test.utils.isSimilarTo
 import coil3.test.utils.size
 import kotlin.test.assertEquals
+import kotlin.test.assertFails
 import kotlin.test.assertFailsWith
 import kotlin.test.assertFalse
 import kotlin.test.assertIs
 import kotlin.test.assertTrue
+import kotlin.time.Duration
 import kotlin.time.Duration.Companion.minutes
+import kotlinx.coroutines.test.TestResult
 import kotlinx.coroutines.test.runTest
 import okio.buffer
 import okio.fakefilesystem.FakeFileSystem
 import okio.source
 import org.junit.Test
 
-class BitmapFactoryDecoderTest {
+private fun interface DecoderTestScope {
+    fun factory(): Decoder.Factory
+}
+
+class AndroidDecoderTest {
```

Comment:
Let's use JUnit 4 parameterized tests for the decoder tests: https://github.com/junit-team/junit4/wiki/parameterized-tests

Reply:
Done